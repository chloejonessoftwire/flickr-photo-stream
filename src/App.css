@import url('https://fonts.googleapis.com/css?family=Open+Sans|Roboto|Comfortaa|Shadows+Into+Light');
body{
  font-family: "Open Sans", sans-serif;
}
h1{
  font-family: 'Comfortaa', serif;
}
h4{
  font-family: 'Shadows Into Light', sans-serif;
}
p {
font-family: "Roboto", sans-serif;
}
#home{
  background-image: linear-gradient(white, #ffeeba21, #e8daff2b,#e8daff47,#438dbd,#438dbd,#438dbd, #3a7dab);
}

.form-inline{
  display:flex;
  justify-content: flex-end;
  padding-right: 20px;
}
.form-control{
  width: 30%;
  max-height: 38px; 
}
.header{
  text-align: center;
}
.header h1{
  color:navy;
  text-shadow: 1px 1px #cacaca;
  margin-top: 50px;
}
.header h4{
  font-size: large;
  font-weight: 400;
  margin-bottom: 0px;

}
.grid-container{
 width: 100%;
 justify-content: space-around;
 column-gap: 0px;
 text-align: center;
}
.image-post{ 
  background: white;
  border-radius: 5px;
  display: inline-block;
  border: 1px solid rgba(116, 116, 116, 0.74);
  margin: 20px;
  box-shadow: 0 4px 8px 0 rgba(46, 38, 38, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19); 
  width: 90%; 
}
.image-post:hover{
  width: 91%;
  border: 5px solid rgb(255, 133, 34);


}
.image{
  display: block;
  margin: auto;
  margin-top: 5%;
  margin-bottom: 10px;
  border: 1px solid black;
  width: 90%;
}
.title-author{
  display:block;
  justify-content: space-between;
  flex: wrap;
  padding: 5px;
}
.title-author h3{
  margin: 0px;
}
.title-author a{
  color:#0078ca;
  text-decoration: none;
}
.title-author a:hover{
  color:#7549c7;
  text-decoration: underline;
}
.title{
  font-size: small;
  text-align: left;
}
.author{
  font-size: smaller;
  text-align: right;
}
.description{
  font-size: small;
  padding: 5px;
  text-align: left;
}
.description p{
  margin: 0px;
}
.tags{
  font-size: smaller;
  color: darkslategrey;
  text-transform: capitalize;
  padding: 8px;
  padding-top: 0px;
  text-align: justify;
}

.footer-image{
  width: 120%;
  margin-left: -10%;
  margin-right: -10%;
  margin-bottom: -10px;
  filter: brightness(120%) saturate(60%) contrast(80%) opacity(80%);
  
}
@media screen and (min-width: 300px) and (max-width: 660px) {
  .grid-container{
      column-count: 1;
  }
  .image-post{
      min-width: 300px;
      max-width: 660px;
  }
  .form-control{
      width: 60%;
      margin-top: .5rem!important;
      margin-bottom: .5rem!important;
  }
  .form-inline{
      display:flex;
      justify-content: center;
      padding-right: 0px;
  }
}
@media screen and (min-width: 660px)  and (max-width: 1000px) {
  .grid-container{
      column-count: 2;
  }
  .image-post{
      min-width: 300px;
      max-width: none;
  }
}
@media screen and (min-width: 1000px)  and (max-width: 1400px) {
  .grid-container{
      column-count: 3;
  }
}
@media screen and (min-width: 1400px)and (max-width: 1900px)  {
  .grid-container{
      column-count: 4;    
  }
}
@media screen and (min-width: 1900px) and (max-width: 2400px) {
  .grid-container{
      column-count: 5;    
  }
}
@media screen and (min-width: 2400px) {
  .grid-container{
      column-count: 6;    
  }
}