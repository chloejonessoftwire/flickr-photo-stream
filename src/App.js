import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor(){
    super();
    this.state = {
      pictures: [],
    };
  }

  componentDidMount(){
    fetch('https://api.flickr.com/services/rest/?method=flickr.photos.search&api_key='+process.env.REACT_APP_API_KEY+'&format=json&lang?en&safe_search=1&tags=derbyshire&extras=owner_name,url_s,url_m,url_l,date_taken,description,tags&nojsoncallback=1')
    .then(function(response){
      return response.json();
    })
    .then(function(data){
      let picArray = data.photos.photo.map((pic) => {
        var title =pic.title;
        var media = 'https://farm'+pic.farm+'.staticflickr.com/'+pic.server+'/'+pic.id+'_'+pic.secret+'.jpg';
        var date= new Date(pic.datetaken).toLocaleDateString('en-GB', {weekday: 'long', day: 'numeric', month: 'long', year: 'numeric'});
        var time = new Date(pic.datetaken).toLocaleTimeString('en-GB',{ hour: 'numeric',minute: 'numeric', hour12: true });
        var author = pic.ownername;
        if (pic.tags==''){
          var tags = ''
        }
        else{
          var tags=(pic.tags.split(' ')).join(', #');
        }
        var link = 'https://www.flickr.com/photos/'+pic.owner+'/'+pic.id+'/';
        var author_link='https://www.flickr.com/photos/'+pic.owner+'/';

        return(
          <div class='image-post'>
            <img class='image' src={media} alt={title}/>
            <div class='title-author'>
              <h3 class='title'> <a href={link} target='_blank'>{title} </a></h3>
              <h3 class='author'> by <a href={author_link} target="_blank">{author}</a></h3>
            </div>
            <div class='description'>
              <p> Posted on: {date} at {time}. </p>
            </div>
            <div class='tags'> 
                <p> Tags:  #{tags}</p>
            </div>
           </div>
          
        )
      })
      this.setState({pictures: picArray});
    }.bind(this))
    .catch((err) => {
      console.log(err)
      })
  }

  render() {
    return (

      <body id='home'>
        <div>
          <div class='header'>
            <h1>Flickr Photo Stream</h1>
            <h4> by Chloe Jones</h4>
          </div>
        {/* <div class='search-bar'>
          <form class="form-inline my-2 my-lg-0">
            <input class="form-control mr-sm-2" type="search" placeholder="Search flickr for photos" aria-label="Search">
            <button class="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
          </form>
        </div> */}
          <br/>
          <div class='grid-container'>
          {this.state.pictures}
          </div>
          <div class='footer'>
            <img class='footer-image' src='https://peakdistrictwalks.net/wp-content/uploads/2020/06/Bamford-Edge-Peak-District-70.jpg'/>
          </div>
        </div>
      </body>
    );
  }
}

export default App;