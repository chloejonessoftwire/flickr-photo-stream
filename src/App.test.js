import { fireEvent, render, waitFor, screen, toBeTruthy, getByTestId, cleanup} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {shallow} from 'enzyme';
import App from './App';
import React from 'react';



test('renders Flick Photo Screen Title', () => {
  render(<App />);
  const linkElement = screen.getByText(/Flickr Photo Stream/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Unit Tests for App', () => {
  test('Test Rendering', () => {
    const {getByTestId} = render(<App/>)
    expect(getByTestId('content')).toBeInTheDocument()
  })
  test('Search bar input should render', () => {
    const {getByTestId} = render(<App/>)
    const searchInput = getByTestId('search-bar-input');
    expect(searchInput).toBeInTheDocument()
    expect(searchInput).toHaveAttribute("type", "search");
    // fireEvent.change(searchInput, { target: { value: 'arizona' } })
    // expect(this.state.search).toBe('arizona')
  })

  test('Derbyhire is in the document (flickr has returned the right json)', async () => {
    render(<App />);
    await waitFor(() => expect(screen.findByText(/derbyshire/i)).toBeInTheDocument);
    // const post = screen.getByTestId("image-post");
    // expect(post).toBeInTheDocument();
    })

    
  test('Pictures are loading', async () => {
    render(<App />);
    await waitFor(() => expect(screen.findByTestId('image-post')).toBeInTheDocument);
    })

  test('Search bar input should change state and load new pages', async () => {
    const {getAllByTestId} = render(<App/>)
    const input = getAllByTestId('search-bar-input')
    fireEvent.change(input[0], {target:{ value: 'arizona'}});

    render(<App />);
    await waitFor(() => expect(screen.findByText(/arizona/i)).toBeInTheDocument);
  })

});

afterEach(cleanup)
 
it('should take a snapshot', () => {
   const { asFragment } = render(<App />)
   
   expect(asFragment(<App />)).toMatchSnapshot()
  })