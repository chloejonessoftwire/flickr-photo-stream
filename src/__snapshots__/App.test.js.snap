// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="content"
  >
    <div
      class="header"
    >
      <h1>
        Flickr Photo Stream
      </h1>
      <h4>
         by Chloe Jones
      </h4>
    </div>
    <div
      class="search-bar"
    >
      <form
        class="form-inline my-2 my-lg-0"
      >
        <input
          aria-label="Search"
          class="form-control mr-sm-2"
          data-testid="search-bar-input"
          placeholder="Search photos"
          type="search"
        />
        <button
          class="btn btn-outline-success my-2 my-sm-0"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <br />
    <div
      class="grid-container"
    />
    <div
      class="footer"
      id="footer"
    >
      <img
        class="footer-image"
        src="https://peakdistrictwalks.net/wp-content/uploads/2020/06/Bamford-Edge-Peak-District-70.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
